CIPs/cip-0051.md

```diff
@@ -0,0 +1,155 @@
+- operators experience churn in needing to upgrade their services
+- monitoring and observing the performance of the system overall is complicated, and information is largely incomplete
+
+ASv2 aims to improve verification rates, improve the wallet and end consumer experience, increase adoption of identity mappings, and encourage network effects of shared cross-application identity attestations.
+
+## Specification
+
+Under our proposal, individual wallets or other entities will act as issuers to implement verification flows and issue credentials attesting to ownership over users' identifiers (phone numbers, email addresses, social media accounts, etc.). _It is ultimately up to issuers to determine how to confirm that a user owns both a particular on-chain address and an off-chain identifier._
+
+Ensuring interoperability of attestations across the ecosystem is a key component of our vision. Currently, phone number identifier → address mappings (attestations) are stored in `Attestations.sol`. Anyone can query ODIS for the phone number identifier, in order to perform the on-chain lookup of stored attestations, but the reverse is not true: phone number attestations that are not created through integrating with ASv1 are not currently discoverable from other dApps. The proposed Federated Attestation Protocol aims to reconcile the benefits of dApps having full control over their verification flow with the benefits of on-chain interoperability. Alongside `Attestations.sol` we implement a new on-chain general attestation registry `FederatedAttestations.sol`, which will allow anyone to issue and publish their own identifier ↔ address mappings. While any account can write to the registry contract, wallets/dApps can choose to read only mappings from issuers they trust.
+
+## Implementation
+
+### Identifiers
+
+While issuers can choose to use identifiers in any format, we recommend using ODIS to derive identifiers that will be published on-chain. This offers a privacy-preserving standard that also allows for discoverability and interoperability. ODIS will be upgraded with the release of ASv2 to include prefix types beyond phone numbers to accomodate all identifier types.
+
+In the future, we plan to allow for issuers to create attestations for the fact that a given identifier was derived using the “canonical” method described above, through the addition of metadata attestations. Identifiers derived using this “canonical” method could be used as a lightweight sybil resistant signal, as there is a one-to-one mapping between off-chain identities (such as phone numbers) and the derived identifier.
+
+### Issuers
+
+Issuers should create a separate signer key that is associated with their account for the express purpose of signing verifiable credentials. The `AttestationSigner` role should be assigned to this signer key. It is possible to authorize multiple signer keys for this role.
```

Comment:
Possible typo: 

> for the express purpose of signing verifiable credentials

Presumably to sign "attestations" instead of "verifiable credentials"?